### DISCUSSION TITLE
How should NFTypes and nfInstanceID be defined and assigned in CMPv2 for both 3GPP-specified NFs and proprietary NFs?

### DISCUSSION THREAD

**Discussion by pragya1998hal** [Keyfactor/ejbca-ce#735] [General]
**Questions:**  
- We have predefined NFTypes for network functions (NFs) such as AMF, UPF, SMF, etc., as outlined in the 3GPP specifications. How should we define and assign `nfInstanceID` for these NFs and their microservices?  
- In addition to the standard NFs, we have proprietary NFs along with their respective microservices. How should we define and assign `nfInstanceID` or `NFType` for these proprietary NFs and their microservices?  
- Is it feasible to use a POD or container ID as the `nfInstanceID` for both 3GPP-specified NFs and proprietary NFs, including their microservices? How would this approach handle scenarios involving microservice replicas?

**Comment by pragya1998hal** [comment]
Hello sir please help me by answering my questions

**Reply by primetomas** [reply]
This honestly looks more like a use case professional services question than an EJBCA specific question. We have people that have worked for the telco industry a lot, but you will not find their answers here. We can answer technical questions on EJBCA, but typically not be able to answer use case specific solution architecture. 
Unless someone in the Community happened to have done a similar thing recently and are willing to share their experience.

**Comment by pragya1998hal** [comment]
ok sir
